tealium/collectors/attribution/TealiumAdClient.swift

```diff
@@ -24,8 +25,145 @@ public class TealiumAdClient: TealiumAdClientProtocol {
 
     }
 
-    public func requestAttributionDetails(_ completionHandler: @escaping ([String: NSObject]?, Error?) -> Void) {
-        adClient.requestAttributionDetails(completionHandler)
+    func getAAAttributionData(details: [String: NSObject]?) -> PersistentAttributionData {
+        var appleAttributionDetails = PersistentAttributionData()
+        if let conversionType = details?[AppleInternalKeys.AAAAttribution.conversionType.rawValue] as? String {
+            appleAttributionDetails.conversionType = conversionType
+        }
+        if let clickedWithin30D = details?[AppleInternalKeys.attribution] as? String {
+            appleAttributionDetails.clickedWithin30D = clickedWithin30D
+        }
+        if let clickedDate = details?[AppleInternalKeys.AAAAttribution.clickDate.rawValue] as? String {
+            appleAttributionDetails.clickedDate = clickedDate
+        }
+        if let keywordId = details?[AppleInternalKeys.AAAAttribution.keywordId.rawValue] as? String {
+            appleAttributionDetails.adKeyword = keywordId
+        }
+        if let orgId = details?[AppleInternalKeys.AAAAttribution.orgId.rawValue] as? String {
+            appleAttributionDetails.orgId = orgId
+        }
+        if let countryOrRegion = details?[AppleInternalKeys.AAAAttribution.countryOrRegion.rawValue] as? String {
+            appleAttributionDetails.region = countryOrRegion
+        }
+        if let adGroupId = details?[AppleInternalKeys.AAAAttribution.adGroupId.rawValue] as? String {
+            appleAttributionDetails.adGroupId = adGroupId
+        }
+        if let campaignId = details?[AppleInternalKeys.AAAAttribution.campaignId.rawValue] as? String {
+            appleAttributionDetails.campaignId = campaignId
+        }
+        if let adId = details?[AppleInternalKeys.AAAAttribution.adId.rawValue] as? String {
+            appleAttributionDetails.adId = adId
+        }
+        return appleAttributionDetails
+    }
+
+    // swiftlint:disable cyclomatic_complexity
```

Comment:
Here instead of having to disable the cyclomatic_complexity we could change the body of the function, instead of having a bunch of `if let` we could have directly something like:
```
        appleAttributionDetails.clickedWithin30D = detailsDict[AppleInternalKeys.attribution] as? String
```
For all of the parameters. 
This applies to the `getAAAttributionData` method too.